tests/unit/reports/multi_table/test_multi_table_quality_report.py

```diff
@@ -22,610 +22,257 @@ def test___init__(self):
-            'Column Pair Trends': [corr_sim_mock, cont_sim_mock],
-            'Cardinality': [cardinality_mock],
-        }
+    def test_get_visualization_missing_table_name(self):
+        """Test it when table_name is missing and property is not Cardinality."""
+        # Setup
+        report = QualityReport()
+        report._is_generated = True
+        report._properties_instances = {'Column Shapes': None}
+        report._tables = ['tab1']
 
-        # Run
-        with patch.object(
-            QualityReport,
-            'METRICS',
-            metrics_mock,
-        ):
-            report = QualityReport()
-            report.generate(real_data, synthetic_data, metadata)
+        # Run and Assert
+        msg = "Table name must be provided when viewing details for property 'Column Shapes'."
+        with pytest.raises(ValueError, match=msg):
+            report.get_visualization('Column Shapes')
 
-        # Assert
-        ks_complement_mock.compute_breakdown.assert_called_once_with(
-            real_data, synthetic_data, metadata)
-        tv_complement_mock.compute_breakdown.assert_called_once_with(
-            real_data, synthetic_data, metadata)
-        corr_sim_mock.compute_breakdown.assert_called_once_with(
-            real_data, synthetic_data, metadata)
-        cont_sim_mock.compute_breakdown.assert_called_once_with(
-            real_data, synthetic_data, metadata)
-        cardinality_mock.compute_breakdown.assert_called_once_with(
-            real_data, synthetic_data, metadata)
-
-        assert report._overall_quality_score == 0.43333333333333335
-        assert report._property_breakdown == {
-            'Column Shapes': 0.15000000000000002,
-            'Column Pair Trends': 0.15000000000000002,
-            'Cardinality': 1.0,
-        }
-        assert report._metric_results['TVComplement'] == {}
+    def test_get_visualization_invalid_table_name(self):
+        """Test it when table_name is invalid."""
+        # Setup
+        report = QualityReport()
+        report._is_generated = True
+        report._properties_instances = {'Column Shapes': None}
+        report._tables = ['table']
 
-    def test_get_score(self):
-        """Test the ``get_score`` method.
+        # Run and Assert
+        msg = re.escape("Unknown table ('invalid_table'). Must be one of ['table'].")
+        with pytest.raises(ValueError, match=msg):
+            report.get_visualization('Column Shapes', 'invalid_table')
 
-        Expect that the overall quality score is returned.
+    def test_get_details(self):
+        """Test the ``get_details`` method."""
+        # Setup
+        report = QualityReport()
+        instance = Mock()
+        instance._details = {'details'}
+        report._properties_instances = {'Cardinality': instance}
+        report._is_generated = True
 
-        Setup:
-        - Mock the ``_overall_quality_score`` attribute.
+        # Run
+        details = report.get_details('Cardinality')
```

Comment:
we should add a test for when the property is cardinality but they give a table name to make sure the filtering works